packages/next-auth/src/next/index.ts

```diff
@@ -113,6 +113,11 @@ function NextAuth(
   res: NextApiResponse,
   options: AuthOptions
 ): any
+function NextAuth(
```

Comment:
Thanks for this contribution! I worked on something like this recently in #7999, and pulled down your repo to take a look. All looks good, except for this warning:

<img width="764" alt="Screenshot 2023-08-02 at 00 15 39" src="https://github.com/nextauthjs/next-auth/assets/43508242/6cbac103-9cb7-4255-9eb8-664cdaad9be8">

Based on my solution, I think it is because you are missing this type on the function's arguments:
```ts
function NextAuth(
  ...args:
    | [AuthOptions]
    | [NextApiRequest, NextApiResponse, AuthOptions]
    | [NextRequest, any, AuthOptions] // This one is missing
) {
//...
}
```

Reply:
Thanks for the suggestion @rexfordessilfie. I've updated my branch now with the compatible implementation signature.